packages/runtime-core/src/hydration.ts

```diff
@@ -327,7 +327,8 @@ export function createHydrationFunctions(
           for (const key in props) {
             if (
               (forcePatchValue && key.endsWith('value')) ||
-              (isOn(key) && !isReservedProp(key))
+              (isOn(key) && !isReservedProp(key)) ||
+              customElements.get(el.localName) // If the element is a custom element, apply vue props during hydration - See #7203
```

Comment:
You should be able to use the app config function like:
```ts
parentComponent.appContext.config.isCustomElement?.(el.localName)
```
That would be more desirable as we don't want to reference any DOM apis in `runtime-core`.

That check should also be done sooner, outside of the for loop.

Reply:
In the test app we're working with (nuxt 3) the `isCustomElement` function is always undefined, so we've not been able to use that function. If you're certain that it should be there, I'll dig into where it goes missing some more

Reply:
My apologies, it's:

```js
parentComponent.appContext.config.compilerOptions.isCustomElement?.(el.localName)
```
See https://vuejs.org/guide/extras/web-components.html#using-custom-elements-in-vue

And of course, that function has to be added by the developer.